### DISCUSSION TITLE
how to get Task inputs from TaskStartedEventArgs?

### DISCUSSION THREAD

**Discussion by meirumeiru** [dotnet/msbuild#8248] [Q&A] [answered]
I found the "IEventSource3::IncludeTaskInputs" and I see that selecting "diagnose" level for compilations in Visual Studio does in fact log inputs of tasks, but I have no idea how to find the inputs of tasks from within a TaskStartedEventHandler. What's the purpose of "IncludeTaskInputs"? How does it work? Where are those inputs then? How can I access them?

**Comment by rainersigwald** [comment, marked as answer]
They are logged as separate `TaskParameterEventArgs` instances, not directly in the `TaskStartedEvent`. And there's not (currently, ask if it would help :)) a separate event for them, so you would have to listen to `AnyEventRaised`, then filter. An example of that can be found in our binary logger:

https://github.com/dotnet/msbuild/blob/eb5ebc23a99af31731a6b6c403ceda0b44aa9966/src/Build/Logging/BinaryLogger/BinaryLogger.cs#L184

https://github.com/dotnet/msbuild/blob/eb5ebc23a99af31731a6b6c403ceda0b44aa9966/src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs#L432

https://github.com/dotnet/msbuild/blob/eb5ebc23a99af31731a6b6c403ceda0b44aa9966/src/Build/Logging/BinaryLogger/BuildEventArgsWriter.cs#L524-L536

**Reply by meirumeiru** [reply]
thanks a lot for this...
but... now I see, that I still don't get those TaskParameterEventArgs... I don't know if it is enough to call 		((IEventSource3)eventSource).IncludeTaskInputs(); in the Initialize function... do I have to activate them also somewhere else?

**Reply by meirumeiru** [reply]
ah, ok... BuildParameters -> LogTaskInputs that's what I had to set to 'true'
